### PR TITLE
Added support for installing podman

### PR DESCRIPTION
**Pull request by Albert-LGTM**
## Type of Change
- [x] New feature

## Description
I added a new feature to install Podman and Podman Compose in the installation script. 

## Testing
I tested the script on Fedora using `dnf` and Ubuntu with apt and pip3. Podman and Podman Compose were installed and worked as expected.

## Impact
This change allows users to install Podman and Podman Compose for those who wish to use it instead of docker

## Additional Information
podman-compose is only available to the dnf package manger, for other distroes it uses the pip3 package. 

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/podman-setup.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/bin/sh -e
+
+. ../common-script.sh
+
+# Function to prompt the user for installation choice
```
Comment (adamperkowski) [inline core/tabs/applications-setup/podman-setup.sh:5]:
```suggestion
```

File: core/tabs/applications-setup/podman-setup.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            exit 1
+            ;;
+    esac
+}
+
+install_podman_compose() {
+    printf "%b\n" "${YELLOW}Installing Podman Compose...${RC}"
+    case "$PACKAGER" in
+        apt-get|nala|zypper|pacman)
+            "$ESCALATION_TOOL" pip3 install podman-compose
```
Comment (adamperkowski) [inline core/tabs/applications-setup/podman-setup.sh:49]: You need to check for pip and python first. And install of course. And I'm pretty sure `pip3 install` would return an error. Installing system-wide packages with pip is not safe. See https://veronneau.org/python-311-pip-and-breaking-system-packages.html

File: core/tabs/applications-setup/podman-setup.sh (hunk 1 above)
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-setup.sh:5]: please remove this LLM generated comment
```suggestion
```
use of LLMs is prohibited unless the code works and is tested:
https://github.com/ChrisTitusTech/linutil/blob/main/.github/CONTRIBUTING.md

please read the contributing guidelines before opening a pull request

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

**Comment by adamperkowski** [comment]
Please **test your changes** before submitting a PR. Read the [Guidelines](https://github.com/ChrisTitusTech/linutil/blob/main/.github/CONTRIBUTING.md).

**Comment by adamperkowski** [comment]
Also, you need to run `cargo xtask docgen` in the root directory.

File: core/tabs/applications-setup/podman-setup.sh (hunk 1 above)
Comment (Albert-LGTM) [reply core/tabs/applications-setup/podman-setup.sh]: Sorry about that, fixed

File: core/tabs/applications-setup/podman-setup.sh (hunk 2 above)
Comment (Albert-LGTM) [reply core/tabs/applications-setup/podman-setup.sh]: I changed it to a user-level installation for Podman Compose with pip3 and updated the PATH variable to include the user's local bin directory in order to use podman-compose installed with pip3. Tested on Ubuntu and Fedora

File: core/tabs/applications-setup/podman-setup.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    fi
+}
+
+install_podman_compose() {
+    printf "%b\n" "${YELLOW}Installing Podman Compose...${RC}"
+    case "$PACKAGER" in
+        apt-get|nala|zypper|pacman)
+            pip3 install --user podman-compose
+            export PATH="$HOME/.local/bin:$PATH"
+            ;;
```
Comment (adamperkowski) [inline core/tabs/applications-setup/podman-setup.sh:96]: Still doesn't solve this:
image
`podman-compose` is available in Arch's `extra` repo. No need to use `pip`.
image

**Comment by adamperkowski** [comment]
`podman-compose` is available as a package in arch/extra ([here](https://archlinux.org/packages/extra/x86_64/podman), [here](https://archlinux.org/packages/extra/any/podman-compose)), debian >= 12 (so also ubuntu) with `apt` and openSUSE >= Leap 15 with `zypper`. Took me one quick web search. Don't use `pip`.

**Comment by Albert-LGTM** [comment]
I have updated the installation method to use the package manager instead of `pip`. I am sorry for  overlooking the package manager option.

I also wanted to clarify that the LLM comment was derived from the `docker-setup.sh` which my script is based upon.

File: core/tabs/applications-setup/podman-setup.sh [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    printf "%b\n" "${YELLOW}Installing Podman...${RC}"
+    case "$PACKAGER" in
+        apt-get|nala)
+            "$ESCALATION_TOOL" "$PACKAGER" install -y podman
+            ;;
+        zypper)
+            "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install podman
+            ;;
+        pacman)
+            "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm podman
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-setup.sh:32]:
```suggestion
            "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm --needed podman
```

File: core/tabs/applications-setup/podman-setup.sh [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    printf "%b\n" "${YELLOW}Installing Podman Compose...${RC}"
+    case "$PACKAGER" in
+        apt-get|nala)
+            "$ESCALATION_TOOL" "$PACKAGER" install -y podman-compose
+            ;;
+        zypper)
+            "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install podman-compose
+            ;;
+        pacman)
+            "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm podman-compose
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-setup.sh:54]:
```suggestion
            "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm --needed podman-compose
```

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

File: core/tabs/applications-setup/podman-setup.sh [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+. ../common-script.sh
+
+choose_installation() { 
+    clear
+    printf "%b\n" "${YELLOW}Choose what to install:${RC}"
+    printf "%b\n" "1. ${YELLOW}Podman${RC}"
+    printf "%b\n" "2. ${YELLOW}Podman Compose${RC}"
+    printf "%b\n" "3. ${YELLOW}Both${RC}"
+    printf "%b" "Enter your choice [1-3]: "
+    read -r CHOICE
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-setup.sh:12]: separate podman and podman-compose into separate scripts, Chris does not like interactive scripts

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

**Comment by adamperkowski** [comment]
> I have updated the installation method to use the package manager instead of `pip`. I am sorry for overlooking the package manager option.

No problem.

> I also wanted to clarify that the LLM comment was derived from the `docker-setup.sh` which my script is based upon.

Well, that's unfortunate. I wonder who "wrote" docker setup...... @nnyyxxxx ??

File: core/tabs/applications-setup/podman-setup.sh (hunk 6 above)
Comment (Albert-LGTM) [reply core/tabs/applications-setup/podman-setup.sh]: Are my latest commits acceptable in resolving this issue?

File: core/tabs/applications-setup/podman-compose-setup.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    esac
+}
+
+install_components() {
+    if ! command_exists podman-compose || ! command_exists podman compose version; then
+        install_podman_compose
+    else
+        printf "%b\n" "${GREEN}Podman Compose is already installed.${RC}"
+    fi
+}
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-compose-setup.sh:33]:
```suggestion
installPodmanCompose() {
    if ! command_exists podman-compose; then
        printf "%b\n" "${YELLOW}Installing Podman Compose...${RC}"
        case "$PACKAGER" in
            apt-get|nala)
                "$ESCALATION_TOOL" "$PACKAGER" install -y podman-compose
                ;;
            zypper)
                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install podman-compose
                ;;
            pacman)
                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm --needed podman-compose
                ;;
            dnf)
                "$ESCALATION_TOOL" "$PACKAGER" install -y podman-compose
                ;;
            *)
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
                exit 1
                ;;
        esac
    else
        printf "%b\n" "${GREEN}Podman Compose is already installed.${RC}"
    fi
}
```

File: core/tabs/applications-setup/podman-compose-setup.sh [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+checkEnv
+checkEscalationTool
+install_components
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-compose-setup.sh:37]:
```suggestion
install_podman_compose
```

File: core/tabs/applications-setup/podman-setup.sh [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    esac
+}
+
+install_components() {
+    if ! command_exists podman; then
+        install_podman
+    else
+        printf "%b\n" "${GREEN}Podman is already installed.${RC}"
+    fi
+}
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-setup.sh:33]:
```suggestion
installPodman() {
    if ! command_exists podman; then
        printf "%b\n" "${YELLOW}Installing Podman...${RC}"
        case "$PACKAGER" in
            apt-get|nala)
                "$ESCALATION_TOOL" "$PACKAGER" install -y podman
                ;;
            zypper)
                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install podman
                ;;
            pacman)
                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm --needed podman
                ;;
            dnf)
                "$ESCALATION_TOOL" "$PACKAGER" install -y podman
                ;;
            *)
                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
                exit 1
                ;;
        esac
    else
        printf "%b\n" "${GREEN}Podman is already installed.${RC}"
    fi
}
```

File: core/tabs/applications-setup/podman-setup.sh [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+checkEnv
+checkEscalationTool
+install_components
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/podman-setup.sh:37]:
```suggestion
install_podman
```

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

**Comment by nnyyxxxx** [comment]
click commit on these new suggestions and this Pr should be ready to go! @Albert-LGTM

**Review by nnyyxxxx — APPROVED** [review]

File: core/tabs/applications-setup/podman-compose-setup.sh [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install podman-compose
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm --needed podman-compose
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y podman-compose
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/podman-compose-setup.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ${PACKAGER}${RC}"
```

File: core/tabs/applications-setup/podman-setup.sh [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" --non-interactive install podman
+                ;;
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm --needed podman
+                ;;
+            dnf)
+                "$ESCALATION_TOOL" "$PACKAGER" install -y podman
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
```
Comment (jeevithakannan2) [inline core/tabs/applications-setup/podman-setup.sh:22]:
```suggestion
                printf "%b\n" "${RED}Unsupported package manager: ${PACKAGER}${RC}"
```

**Review by jeevithakannan2 — CHANGES_REQUESTED** [review]

**Review by adamperkowski — APPROVED** [review]

**Pull request merged**